src/components/rating/rating.tsx

```diff
@@ -0,0 +1,183 @@
+   */
+  @Event() input: EventEmitter;
+
+  onClick(event: UIEvent) {
+    const element = event.target as HTMLElement;
+    const targetParent = element.parentElement;
+    const score =
+      element.nodeName === "polygon"
+        ? Array.from(targetParent.parentElement.children).indexOf(
+            targetParent
+          ) + 1
+        : Array.from(targetParent.children).indexOf(element) + 1;
+    this.value = score;
+    this.input.emit(this);
+  }
+
+  /**
+   * Returns the id of the inner `input` element (if set).
+   */
+  @Method()
+  async getNativeInputId() {
+    return this.element.querySelector("input").id;
+  }
+
+  private renderStarsRating() {
+    const stars = [];
+    for (let i = 0; i < 5; i++) {
+      stars.push(
+        <svg
+          class="rating"
+          {...this.svgViewport}
+          onClick={ev => this.onClick(ev)}
+        >
+          {this.starShape}
+        </svg>
+      );
+    }
+    return stars;
+  }
+  private renderStarsScore() {
+    const stars = [];
+    let percent: number;
+    let starsScore: number;
+    if (this.maxValue - this.value >= 0) {
+      percent = (this.value * 100) / this.maxValue;
+      starsScore = Math.round((percent * 5) / 100);
+    } else {
+      // tslint:disable-next-line:no-console
+      console.error("'value' cannot be greater than 'max-value'");
+    }
+    for (let i = 0; i < 5; i++) {
+      stars.push(
+        <svg
+          class={{
+            active: i < starsScore,
+            score: true
+          }}
+          {...this.svgViewport}
+        >
+          {this.starShape}
+        </svg>
+      );
+    }
+    return stars;
+  }
+
+  render() {
+    const valuesDifference = this.maxValue - this.value;
+    if (!this.inputId) {
+      this.id
+        ? (this.inputId = `${this.id}_inputRange`)
+        : (this.inputId = `gx-inputRange-auto-id-${autoInputRangeId++}`);
+    }
+    if ((valuesDifference >= 0 && this.readonly) || !this.readonly) {
+      return (
+        <div>
+          <input
+            id={this.inputId}
+            type="range"
+            min="0"
+            max={this.readonly ? this.maxValue : 5}
+            step="1"
+            value={this.value}
+          />
+          <div
+            class={{
+              rating: !this.readonly,
+              score: this.readonly
+            }}
+            data-score={this.value !== 0 ? this.value : undefined}
+          >
+            {this.readonly ? this.renderStarsScore() : this.renderStarsRating()}
+          </div>
+        </div>
+      );
+    } else {
+      if (!!this.maxValue) {
+        // tslint:disable-next-line:no-console
+        console.error(
+          "Incongruence between values. 'value' is higher than 'max-value'.",
+          this
+        );
+      } else {
+        // tslint:disable-next-line:no-console
+        console.error("'max-value' has not a value setted ", this);
```

Comment:
```suggestion
        console.error("'max-value' has not a value set ", this);
```

Reply:
Reflecting on this log messages, I'm wondering if we shouldn't be adding the element to the log line, so the user has context as to where the error was originated.

Reply:
Ready!
It will print the gx element from the DOM